renderer/src/features/registry-servers/components/form-run-from-registry.tsx

```diff
@@ -275,104 +328,175 @@ export function FormRunFromRegistry({
                 installation.
               </DialogDescription>
             </DialogHeader>
-
-            <div className="relative max-h-[65dvh] space-y-4 overflow-y-auto px-6">
-              <FormField
-                control={form.control}
-                name="serverName"
-                render={({ field }) => (
-                  <FormItem className="mb-10">
-                    <FormLabel>Server Name</FormLabel>
-                    <FormDescription>
-                      Choose a unique name for this server instance
-                    </FormDescription>
-                    <FormControl>
-                      <Input
-                        {...field}
-                        placeholder="e.g. my-custom-server"
-                        autoFocus
+            {isSubmitting && (
+              <div className="relative space-y-4 px-6">
+                <Alert>
+                  <Loader className="size-4 animate-spin" />
+                  <AlertTitle>
+                    {isPendingSecrets
+                      ? 'Creating Secrets...'
+                      : 'Installing server...'}
+                  </AlertTitle>
+                  <AlertDescription>
+                    {isPendingSecrets && loadingSecrets
+                      ? loadingSecrets?.text
+                      : 'We are pulling the server image from the registry and installing it.'}
+                    {isPendingSecrets && loadingSecrets && (
+                      <Progress
+                        value={
+                          (loadingSecrets?.completedCount /
+                            loadingSecrets?.secretsCount) *
+                          100
+                        }
+                        className="my-2 w-full"
                       />
-                    </FormControl>
-                    <FormMessage />
-                  </FormItem>
+                    )}
+                  </AlertDescription>
+                </Alert>
+              </div>
+            )}
```

Comment:
nit: I would put it into another file for better readability having only a ternary operator with the below content

Reply:
yeah I did export a dedicated component